packages/typespec-ts/test/modularUnit/scenarios/models/deserialization/discriminatedUnion.md

```diff
@@ -0,0 +1,91 @@
+# Should generate serializers for discriminated model
+
+Verify that the serializers are correctly referenced within the switch statement of the base serializers.
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+@discriminator("kind")
+model AWidgetData {
+    kind: string;
+}
+
+model AOAIModelConfig extends AWidgetData {
+  kind: "kind0";
+  fooProp: string;
+}
+
+model MAASModelConfig extends AWidgetData {
+  kind: "kind1";
+  start: utcDateTime;
+  end?: utcDateTime;
+}
+
+@route("/serialize")
+interface D {
+  @get op bar(): AWidgetData;
+}
+```
+
+## Provide generated models and its serializer
+
+Generated Models.
+
+```ts models
+/** model interface AWidgetData */
+export interface AWidgetData {
+  kind: string;
+}
+
+export function aWidgetDataDeserializer(item: any): AWidgetData {
+  return {
+    kind: item["kind"]
+  };
+}
+
+/** Alias for AWidgetDataUnion */
+export type AWidgetDataUnion = AoaiModelConfig | MaasModelConfig | AWidgetData;
+
+export function aWidgetDataUnionDeserializer(item: any): AWidgetDataUnion {
+  switch (item.kind) {
+    case "kind0":
+      return aoaiModelConfigDeserializer(item as AoaiModelConfig);
```

Comment:
I would assume this is already fixed in my pr: https://github.com/Azure/autorest.typescript/pull/3102.

Reply:
I think the issue is still present there, normalizing twice the subTypeName seems to be the cause of the inconsistency. The long term fix though should be using the binder and resolveReference. But wanted to do the quick fix to unblock our partners.